web-registry/src/lib/wallet/hooks/useDetectKeplrMobileBrowser.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react';
+
+import { Wallet } from '../wallet';
+import { AUTO_CONNECT_WALLET_KEY } from '../wallet.constants';
+import { WalletType } from '../walletsConfig/walletsConfig.types';
+import { ConnectWalletType } from './useConnectWallet';
+
+type Props = {
+  wallet: Wallet;
+  loaded: boolean;
+  connectWallet: ConnectWalletType;
+};
+
+// This hooks handle the keplr native app integration connection
+export const useDetectKeplrMobileBrowser = ({
+  wallet,
+  connectWallet,
+  loaded,
+}: Props): void => {
+  useEffect(() => {
+    if (typeof window === 'undefined' || !loaded || wallet.address !== '') {
+      return;
+    }
+
+    import('@keplr-wallet/stores')
+      .then(({ getKeplrFromWindow }) => getKeplrFromWindow())
+      .then(keplr => {
+        if (keplr && keplr.mode === 'mobile-web') {
+          connectWallet({ walletType: WalletType.Keplr });
```

Comment:
this me just misunderstanding something and looking for clarity. why do we need `WalletType.Keplr` here and not `WalletType.WalletConnectKeplr`? it seems like we're in mobile, so shouldn't we always be using wallet connect in this case?

Reply:
This is intuitive to understand indeed. Basically, when the app will be used within Keplr native app (with their mobile browser) the connection settings will be the same as using the chrome extension.
Will add some comments to make that clear.